### PR TITLE
[Hexagon] Do not pattern match inside if_then_else block

### PR DESCRIPTION
**Pull request by aankit-ca**
Resolves the compilation below compilation error while generating
hannk::upsample_channels_uint8 from hannk/depthwise_conv.generator
for HVX:

Unknown intrinsic dynamic_shuffle

The problem occurs when we pattern match hvx instrinsics inside
if_then_else nodes and try to scalarize them later. In the patch we
prevent matching these intrinsics inside if_then_else blocks.

@dsharletg @pranavb-ca PTAL

### CODE REVIEW

File: src/HexagonOptimize.cpp [hunk 1]
```diff
@@ -914,6 +914,9 @@ class OptimizePatterns : public IRMutator {
     }
 
     Expr visit(const Call *op) override {
+        if (op->is_intrinsic(Call::if_then_else)) {
```
Comment (dsharletg) [inline src/HexagonOptimize.cpp:917]: I think you should only skip this if the condition is a vector. If the condition is a scalar it should be fine and I think might be an important case. Same for the other 4 instances.

Comment (aankit-ca) [reply src/HexagonOptimize.cpp]: Done

Comment (dsharletg) [reply src/HexagonOptimize.cpp]: This is different from what I suggested. This skips it for vector *results*, which are common. The case I am thinking about that we should still match patterns for is if_then_else where the *condition* (the first argument of the call) is a scalar, but the result and operands are vectors.

Comment (aankit-ca) [reply src/HexagonOptimize.cpp]: Oh understood now. I'm assuming we should pattern match where conditions are Broadcasts as well?

Comment (dsharletg) [reply src/HexagonOptimize.cpp]: I think so. I think we might see if_then_else with both scalar or broadcasting conditions, I'd have to poke around in the code to be sure one way or the other.

Comment (aankit-ca) [reply src/HexagonOptimize.cpp]: Yeah. It seems like we can encounter both. Added condition for Broadcasts.

**Review by dsharletg — APPROVED** [review]

**Comment by dsharletg** [comment]
The reason for this change has test coverage in hannk, correct? So once we enable testing of that on Hexagon on the buildbots, this will get coverage?

**Comment by aankit-ca** [comment]
> The reason for this change has test coverage in hannk, correct? So once we enable testing of that on Hexagon on the buildbots, this will get coverage?

Yes

**Pull request merged**